src/Octoshift/Services/AwsApi.cs

```diff
@@ -27,12 +26,8 @@ public AwsApi(string awsAccessKeyId, string awsSecretAccessKey, string awsRegion
 
     private static AmazonS3Client BuildAmazonS3Client(string awsAccessKeyId, string awsSecretAccessKey, string awsRegion, string awsSessionToken)
     {
-        var regionEndpoint = DefaultRegionEndpoint;
-        if (awsRegion.HasValue())
-        {
-            regionEndpoint = GetRegionEndpoint(awsRegion);
-            AWSConfigsS3.UseSignatureVersion4 = true;
-        }
+        var regionEndpoint = awsRegion.IsNullOrWhiteSpace() ? null : GetRegionEndpoint(awsRegion);
```

Comment:
I'd love to cut this out and set the expectation that we always have a region here. During runtime, we always do, because we validate that in `AwsApiFactory`, but it seems that there are some cases in our tests where the `AwsApi` is initialized without going through the factory.

Reply:
the tests (in the process of mocking AwsAPI) will call the constructor with all null args. This is a consequence of the mocking pattern we chose to use, where the mock is technically a subclass of the thing it's mocking. So the constructors need to support being called with all null values (and ideally do a noop in that scenario).

Reply:
I have a workaround but it's a bit sketchy. If you make the 2nd AwsApi ctor public (it's currently internal), then move it so it appears first in the file (because our `TestHelper.CreateMock` uses the first ctor it finds - which we should probably change to use the ctor with the least args) all the tests will pass. This is because the other ctor works fine if you pass it null args (it basically does a noop).

Reply:
Thanks for the heads-up! I've made that change to get this unblocked.